bluebottle/accounting/utils.py

```diff
@@ -0,0 +1,293 @@
+from django.db.models import Sum, Count, Q
+from django.utils import timezone
+from datetime import datetime
+
+from .models import BankTransaction, RemoteDocdataPayment, RemoteDocdataPayout, BankTransactionCategory, BankAccount
+from decimal import Decimal
+
+
+def add_up(first, second):
+    """
+    Addition of two objects.
+
+    Distinguish addition of dictionaries, list and numbers.
+
+    return None when both are None
+    if one of the two is None, return the other one which is not None
+
+    if both are mydicts, add them up with the custom mydict.__add__
+    if both are mylists, add them up with the .add method
+
+    if both are convertable to a Decimal, return the addition of the numbers
+
+    in other cases, return None
+    """
+    if not first:
+        return second
+    elif not second:
+        return first
+    else:
+        if isinstance(first, mydict) and isinstance(second, mydict):
+            return first + second
+        elif isinstance(first, mylist) and isinstance(second, mylist):
+            return first.add(second)
+        else:
+            try:
+                return Decimal(str(first)) + Decimal(str(second))
+            except:
+                return None
+
+
+class mydict(dict):
+
```

Comment:
Dictionary with a custom add method that uses the 'add_up' function to handle all kinds of values (another dict, a list, value, string, None, ..)

When dictA (self) is an empty dict, return dictB (without lookint at the structure)

Only add up two dicts when both have exactly the same keys (return empyty MyDict when they do not have the same keys)

Construct a new MyDict, with exactly the same keys as the two dicts, and with values that are calculated by the add_up function that can handle any kind of nested dicts, lists (of dicts) etc. (this same add_up function is used when each item in two MyLists are added up)